subprojects/workers/src/main/java/org/gradle/workers/internal/DaemonForkOptionsBuilder.java

```diff
@@ -47,7 +56,56 @@ public DaemonForkOptionsBuilder withClassLoaderStructure(ClassLoaderStructure cl
     }
 
     public DaemonForkOptions build() {
-        return new DaemonForkOptions(buildJavaForkOptions(), keepAliveMode, classLoaderStructure);
+        JavaForkOptionsInternal forkOptions = buildJavaForkOptions();
+        if (OperatingSystem.current().isWindows() && keepAliveMode == KeepAliveMode.DAEMON) {
+            List<String> jvmArgs = forkOptions.getAllJvmArgs();
+            if (containsUnreliableOptions(jvmArgs)) {
+                LOGGER.info("Worker requested to be persistent, but its JVM arguments may make the worker unreliable. Worker will expire at the end of the build session.");
```

Comment:
Before merging, we should make a better message here that describes why the worker may be unreliable.  It would be really nice, too, if we provided the command line option that causes it to be considered unreliable.

Reply:
PTAL